lib/puma/server.rb

```diff
@@ -256,7 +256,8 @@ def handle_servers_lopez_mode
           Thread.current.purge_interrupt_queue if Thread.current.respond_to? :purge_interrupt_queue
         end
 
-        @notify.close
+        # Prevent can't modify frozen IOError (RuntimeError)
+        @notify.close rescue nil
```

Comment:
@chwevans I don't know exactly about the fix, but I'm often getting `RuntimeError: can't modify frozen IOError` on Heroku after scaling down.

Reply:
@evanphx sure. I get that almost every day when Heroku [auto restart/recycle dynos](https://devcenter.heroku.com/articles/dynos#automatic-dyno-restarts).

![image](https://user-images.githubusercontent.com/105652/53059204-0e320f80-3484-11e9-88c7-0fba881dc73f.png)